autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -265,6 +267,22 @@ def transform(
         if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
             X = cast(Type[pd.DataFrame], X)
 
+            if self.all_nan_columns is None:
+                raise ValueError('_fit must be called before calling transform')
+
+            for col in list(self.all_nan_columns):
+                X[col] = np.nan
+                X[col] = pd.to_numeric(X[col])
+
+        if len(self.categorical_columns) > 0:
+            if self.column_transformer is None:
+                raise AttributeError("Expect column transformer to be built"
+                                     "if there are categorical columns")
+            categorical_columns = self.column_transformer.transformers_[0][-1]
+            for column in categorical_columns:
+                if X[column].isna().all():
+                    X[column] = X[column].astype('object')
```

Comment:
Change for a huge number of features.
```suggestion
            cat_cols = self.column_transformer.transformers_[0][-1]  # categorical columns
            all_nan_cat_cols = set(X[X[cat_cols].isna().all()].columns())
            dtype_dict = {col: 'object' for col in cat_cols if col in all_nan_cat_cols}
            X = X.astype(dtype_dict)
```